internal/encryption/keyprovider/openbao/provider.go

```diff
@@ -0,0 +1,66 @@
+package openbao
+
+import (
+	"context"
+
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider"
+)
+
+type keyMeta struct {
+	Ciphertext []byte `json:"ciphertext"`
+}
+
+func (m keyMeta) isPresent() bool {
+	return len(m.Ciphertext) != 0
+}
+
+type keyProvider struct {
+	svc       service
+	keyName   string
+	keyLength DataKeyLength
+}
+
+func (p keyProvider) Provide(rawMeta keyprovider.KeyMeta) (keyprovider.Output, keyprovider.KeyMeta, error) {
+	if rawMeta == nil {
+		return keyprovider.Output{}, nil, &keyprovider.ErrInvalidMetadata{
+			Message: "bug: no metadata struct provided",
+		}
+	}
+
+	inMeta, ok := rawMeta.(*keyMeta)
+	if !ok {
+		return keyprovider.Output{}, nil, &keyprovider.ErrInvalidMetadata{
+			Message: "bug: invalid metadata struct type",
+		}
+	}
+
+	ctx := context.Background()
```

Comment:
It might be a good idea add a sensible default timeout here so tofu can't hang indefinitely.

Reply:
Agree, I also want to extend the `keyprovider.Provide` method to pass/receive `context.Context` so timeout could be set earlier. I want to prepare this as another PR (kind of tiny refactoring of mentioned tests and context).

Reply:
Nope.  Contexts in OpenTofu are not well managed.  It's on the TODO list.

Reply:
Just to be clear - I plan to add context passing to the `keyprovider.Provide` call and add the timeout for all the providers so it's planned for a separate PR (to simplify merging since it affects not only Bao).